Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ServerGameNetPortal.cs

```diff
@@ -110,15 +110,19 @@ public void OnUserDisconnectRequest()
             {
                 SendServerToAllClientsSetDisconnectReason(ConnectStatus.HostDisconnected);
                 // wait before shutting down to make sure those messages get sent before the clients disconnect.
-                StartCoroutine(WaitToShutdown());
+                StartCoroutine(WaitToShutdown()); // todo in theory, according to "Shutdown"'s doc, we shouldn't need this wait anymore.
             }
         }
 
         IEnumerator WaitToShutdown()
         {
-            yield return new WaitForSeconds(0.5f);
+            // todo netmanager should really be shutting down instead of us having to manually flag that we're shutting down
+            SceneLoaderWrapper.Instance.IsClosingClients = true;
+            yield return null; // todo still needed? wait for UTP's update for it to send it's batched messages
+            yield return null;

```

Comment:
Too risky for GDC, I presume. Do we have a ticket open to investigate removing these waits?